hbase-client/src/main/java/org/apache/hadoop/hbase/client/ClientServiceCallable.java

```diff
@@ -34,9 +36,14 @@
 public abstract class ClientServiceCallable<T>
   extends RegionServerCallable<T, ClientProtos.ClientService.BlockingInterface> {
 
+  protected Map<String, byte[]> requestAttributes;
+
   public ClientServiceCallable(Connection connection, TableName tableName, byte[] row,
-    RpcController rpcController, int priority) {
-    super(connection, tableName, row, rpcController, priority);
+    RpcController rpcController, int priority, Map<String, byte[]> requestAttributes) {
+    super(connection, tableName, row,
+      HBaseRpcControllerImpl.configureRequestAttributes(rpcController, requestAttributes),
+      priority);
+    this.requestAttributes = requestAttributes;
```

Comment:
can you explain this change?

Reply:
ClientServiceCallable extensions like ScannerCallable and ReversedScannerCallable need this field to be available, so I opted for a protected field on the abstract class (which is inline with the established pattern on the parent RegionServerCallable). That said, not all extensions need it, so we could just add the field to the specific implementations as appropriate

Reply:
I'd still like to know the thoughts behind putting this here, but I think it'd be preferable to just pass the requestAttributes down into the super call here. I wonder why you stopped here? In the super class, a reconfiguration of the controller happens in `call` method. That includes calling `reset()` which doesn't currently reset attributes, but who knows, it may in the future and this would totally get lost. But more i think it just seems like a departure from the pattern, since configuration is happening down one level.

Reply:
Woops, I did not see your response when I added the most recent....

In terms of the reason, why not pipe it down into RegionServerCallable and add as a protected field there? There are other protected fields there, but none in ClientServiceCallable which just seems to be a delegating class of sorts.

Reply:
yeah I think that'd be reasonable, I can make that change